app/assets/javascripts/controllers/middleware_datasource/middleware_datasource_controller.js

```diff
@@ -27,6 +27,12 @@ function MwAddDatasourceCtrl($scope, $rootScope, miqService, mwAddDatasourceServ
       'password': vm.step3DsModel.password,
       'securityDomain': vm.step3DsModel.securityDomain,
     };
+    if (angular.element('#server_id').val()) {
+      payload.id = angular.element('#server_id').val();
```

Comment:
`payload.id` already has `angular.element('#server_id').val()` so no need to assign it once again. I would suggest this:
```javascript
var payload = {...}
if (!angular.element('#server_id').val()) {
   payload.miq_grid_checks = ManageIQ.gridChecks.join(',');
}
return payload;
```